pkg/otelcollector/infra-meter/infra-meter.go

```diff
@@ -31,7 +34,28 @@ func AddInfraMeters(
 		infraMeters = map[string]*policylangv1.InfraMeter{}
 	}
 	for pipelineName, metricConfig := range infraMeters {
-		if err := addInfraMeter(config, pipelineName, metricConfig); err != nil {
+		infraMeterName := strings.Split(pipelineName, "/")[1]
+		processorName := fmt.Sprintf("%s-%s-%s-%d", otelconsts.ProcessorTelemetryCollector, policyName, infraMeterName, telemetryCollectorID)
+		config.AddProcessor(processorName, map[string]any{
+			"attributes": []map[string]interface{}{
+				{
+					"key":    "aperture_infra_meter_name",
+					"action": "upsert",
+					"value":  infraMeterName,
+				},
+				{
+					"key":    metrics.PolicyNameLabel,
+					"action": "upsert",
+					"value":  policyName,
+				},
+				{
+					"key":    "aperture_telemetry_collector_id",
+					"action": "upsert",
+					"value":  telemetryCollectorID,
+				},
+			},
+		})
```

Comment:
Please move adding this processor to `addInfraMeter`

Reply:
@krdln I have added it there earlier but we had to pass all those variables from current function to the `addInfraMeter` function so I kept it here. Are you suggesting to add this processor totally somewhere else?

Reply:
You mean these vars?

> 
> ```
> 	policyName string,
> 	telemetryCollectorID int64,
> ```

Yeah, I think it's cleaner to pass these to `addInfraMeter`. Before all the logic related to the single infra meter was in `addInfraMeter`, but right now a processor is created in `AddInfraMeters` and rest of pipeline in `addInfraMeter` which was confusing to me on the first look

Reply:
Made the changes. Thank you!